### DISCUSSION TITLE
Can .NET Maui Achieve Fast Start-up Time in Release mode like this?

### DISCUSSION THREAD

**Discussion by nextcodelab** [dotnet/maui#13036] [Q&A]
Sample in other frameworks.
It seem pure code no XAML is faster,, no more hundreds of logics and properties. No more bridge to compile native.

https://user-images.githubusercontent.com/73583588/216351217-807a6ec3-151f-4a3d-9cc5-cb7333f6213c.mp4

HERE IS THE .NET MAUI APP IN RELEASE MODE. by https://github.com/rive-app/rive-sharp.

https://user-images.githubusercontent.com/73583588/216381943-09e03ab3-6fd8-4900-b482-1e9a8c950b9b.mp4

I learn this framework just few days ago,, and this framework app development seems  1000x faster than other frameworks.
![Screenshot 2023-01-23 191550](https://user-images.githubusercontent.com/73583588/215914225-cd03901c-868d-43ef-baef-5f23a8846be8.png)

XAML IS GOOD ONLY FOR FIRST PARTY OPERATING SYSTEM, but will be slow on other platforms (third party).
C# means complete C# draw. No more XAML Interpreter or Parser  or compilations for third party OS.

**Comment by Ghevi** [comment]
In my experience even in debug the app isn't so slow to start after the splas schreen, not as fast as flutter tho.
Do you have some long operation in MauiProgram or MainActivity?

**Reply by nextcodelab** [reply]
Maybe flutter is super smooth 400 times than .net Maui.. and development of Flutter I feel 500 times faster... 

My estimate it is because of XAML, lots of compilations and this initial below.


I also dont know if more library to be initialize here will cause the app slow on start up.. but number 1 causes for me is the XAML.
```c#

var builder = MauiApp.CreateBuilder();
            builder
                .UseMauiApp<App>()

                .UseMauiCommunityToolkit()
                .ConfigureFonts(fonts =>
                {
                    fonts.AddFont("OpenSans-Regular.ttf", "OpenSansRegular");
                    fonts.AddFont("FA5Regular.otf", "FA5Regular");
                    fonts.AddFont("GentiumPlusR.ttf", "GentiumPlus");
                    fonts.AddFont("Samantha.ttf", "Samantha");
                    fonts.AddFont("STIXTwoTextSemiBold.ttf", "STIXTwoText");
                    fonts.AddFont("MaterialIcons-Regular.ttf", "Material Icons");
                    fonts.AddFont("fa-solid-900.ttf", "FASolid900");
                });

            builder.Services.AddMauiBlazorWebView();
            builder.Services.AddLocalization();
            builder.Services.AddBlazoredModal();
```

**Reply by shadowfoxish** [reply]
@nextcodelab Have you tried using [Stopwatch](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.stopwatch?view=net-7.0) to time the different parts of your app's startup code to pinpoint where the time is spent?

XAML is supposed to be compiled at build time by default. Please check [here](https://learn.microsoft.com/en-us/dotnet/maui/xaml/xamlc?view=net-maui-7.0) for some reading about that.
